src/gui/wizard/owncloudwizard.cpp

```diff
@@ -112,8 +112,29 @@ OwncloudWizard::OwncloudWizard(QWidget *parent)
 
     // allow Flow2 page to poll on window activation
     connect(this, &OwncloudWizard::onActivate, _flow2CredsPage, &Flow2AuthCredsPage::slotPollNow);
+}
+
+void OwncloudWizard::adjustWizardSizeForPage(QWizardPage *page)
+{
+    // Let the wizard page check how big it is
+    page->adjustSize();
+
+    // Adjust the whole wizard to the biggest size
+    const auto pageSize = page->sizeHint();
+    if (currentBiggestPageSize.width() < pageSize.width()) {
+        currentBiggestPageSize.setWidth(pageSize.width());
+    }
+    if (currentBiggestPageSize.height() < pageSize.height()) {
+        currentBiggestPageSize.setHeight(pageSize.height());
+    }
+    resize(currentBiggestPageSize);
```

Comment:
I'd do that in a loop instead, two reasons: no need to introduce an extra member variable just for that, and no need to call resize as many times as we have pages.

I'd start from calling `pageIds()` then `std::transform` that list into a list of `QSize` and then call `std::max_element` on that new list, which would give the size you need to use with resize.

Reply:
I was looking for function that returns all pages. Thanks.